src/MySqlConnector/Core/ServerSession.cs

```diff
@@ -596,6 +596,63 @@ public async Task DisposeAsync(IOBehavior ioBehavior, CancellationToken cancella
 		return statusInfo;
 	}
 
+	public async Task PostConnectionsQueriesAsync(ConnectionSettings cs,
+		IOBehavior ioBehavior, CancellationToken cancellationToken)
+	{
+		var payloadDatas = new List<PayloadData>();
```

Comment:
The extra allocations here (on a common path) initially concerned me, but I suppose the benefit of pipelining should outweigh the cost of any extra garbage.